internal/buffer/in_message.go

```diff
@@ -52,11 +53,56 @@ func NewInMessage() *InMessage {
 	}
 }
 
+func readAll(r io.Reader, dest []byte) (int, error) {
```

Comment:
How is this function different from io.ReadFull?

Reply:
Perhaps I missed ReadFull() function. I will redo the PR